src/contracts/core/DelegationManager.sol

```diff
@@ -412,46 +531,146 @@ contract DelegationManager is Initializable, OwnableUpgradeable, Pausable, Deleg
+            }
+
+            unchecked { ++i; }
+        }
+
+        // Create queue entry and increment withdrawal nonce
+        uint96 nonce = numWithdrawalsQueued[staker];
+        numWithdrawalsQueued[staker]++;
+
+        Withdrawal memory withdrawal = Withdrawal({
+            staker: staker,
+            delegatedTo: operator,
+            withdrawer: withdrawer,
+            nonce: nonce,
+            startBlock: uint32(block.number),
+            strategies: strategies,
+            shares: shares
+        });
+
+        bytes32 withdrawalRoot = calculateWithdrawalRoot(withdrawal);
+
+        // Place withdrawal in queue
+        pendingWithdrawals[withdrawalRoot] = true;
+
+        emit WithdrawalQueued(withdrawalRoot, withdrawal);
+        return withdrawalRoot;
+    }
+
+    /**
+     * @notice Withdraws `shares` in `strategy` to `withdrawer`. If the shares are virtual beaconChainETH shares, then a call is ultimately forwarded to the
+     * `staker`s EigenPod; otherwise a call is ultimately forwarded to the `strategy` with info on the `token`.
+     */
+    function _withdrawSharesAsTokens(address staker, address withdrawer, IStrategy strategy, uint256 shares, IERC20 token) internal {
+        if (strategy == beaconChainETHStrategy) {
+            eigenPodManager.withdrawSharesAsTokens({
+                podOwner: staker,
+                destination: withdrawer,
+                shares: shares
+            });
+        } else {
+            strategyManager.withdrawSharesAsTokens(withdrawer, strategy, shares, token);
+        }
+    }
+
+    /**
+     * @notice If the shares are virtual beaconChainETH shares, then adds shares to the `staker` and delegates the new shares to the `staker`s operator.
+     * Otherwise adds `shares` in `strategy` to `withdrawer` and delegates them to `operator`
+     */
+    function _addAndDelegateShares(address staker, address withdrawer, address operator, IStrategy strategy, uint256 shares) internal {
+        // When awarding podOwnerShares in EigenPodManager, we need to be sure
+        // to only give them back to the original podOwner. Other strategy shares
+        // can be awarded to the withdrawer.
+        if (strategy == beaconChainETHStrategy) {
+            // update shares amount depending upon the returned value
+            // the return value will be lower than the input value in the case where the staker has an existing share deficit
+            shares = eigenPodManager.addShares({
+                podOwner: staker,
+                shares: shares
+            });
+            // since these shares are given back to the pod owner, they must be delegated to the *pod owner*'s operator, which could be different from the withdrawer's
+            operator = delegatedTo[staker];
+            // we also want to emit the event within `_increaseOperatorShares` with the correct fields. Since the staker receives the shares, they should be in the event.
+            withdrawer = staker;
```

Comment:
Good bug fix here. I think we can simplify this just a touch - currently this method is passed the operator from `completeQueuedWithdrawal`.

However, `completeQueuedWithdrawal` doesn't need the operator for anything -- it's only used in `_addAndDelegateShares`.

Maybe we just don't pass it in, and calculate it in both branches within this function? Probably clearer logic overall, and reduces a storage read for certain cases.

Reply:
Are you opposed to just doing this? https://github.com/Layr-Labs/eigenlayer-contracts/pull/222/commits/0df17301a7f834adb2bbc52e14849adcbdaa5b89
I think it's clearer this way, especially after merging https://github.com/Layr-Labs/eigenlayer-contracts/pull/227 and clarifying the concept of "(change in) delegateable shares"